client/src/pages/constructor/index.ts

```diff
@@ -127,8 +137,23 @@ function removeBoxIsEmptyMessage() {
   boxEmptyMessageElement.classList.add('hidden');
 }
 
-popupHandler();
-tabHandler('forms-container');
-formHandler.addUserButtonListener();
-formHandler.addRegisterFormListener();
-formHandler.addSignInFormListener();
+function addUsersProducts() {
+  const userObject = JSON.parse(localStorage.getItem('user') || 'null') as IUserResponse;
+  const prevAddedProductList = getProductListFromConstructor();
+  if (userObject) {
+    userObject.products.own.forEach(product => {
+      if (!prevAddedProductList.includes(product)) {
+        moveIngredientToChosen(product);
+      }
+    });
+  }
+}
+
+function getProductListFromConstructor(): string[] {
```

Comment:
I would store chosen products list instead of getting them each time from HTML...